src/utils/multihash.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019-2024 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+//!
+//! This module back-fills the Identify hasher and code that was removed in `multihash` crate.
+//! See <https://github.com/multiformats/rust-multihash/blob/master/CHANGELOG.md#-breaking-changes>
+//! and <https://github.com/multiformats/rust-multihash/pull/289>
+//!
+
+pub mod prelude {
+    pub use super::MultihashCodeLegacy;
+    pub use multihash_codetable::{Code as MultihashCode, MultihashDigest as _};
+}
+
+use multihash_derive::MultihashDigest;
+
+#[derive(Clone, Copy, Debug, Eq, MultihashDigest, PartialEq)]
+#[mh(alloc_size = 64)]
+pub enum MultihashCodeLegacy {
```

Comment:
Wow, that's quite a change on their part.

Reply:
Do we then need all of this code, or should we instead introduce a helper method to wrap this: https://github.com/multiformats/rust-multihash/blob/master/examples/identity.rs

Reply:
We do need the code. e.g. `CarBlock::validate`

```rust
pub fn validate(&self) -> anyhow::Result<()> {
        let actual = {
            let code = MultihashCode::try_from(self.cid.hash().code())?;
            Cid::new_v1(self.cid.codec(), code.digest(&self.data))
        };
        anyhow::ensure!(
            actual == self.cid,
            "CID/Block mismatch for block {}, actual: {actual}",
            self.cid
        );
        Ok(())
    }
```